src/clj/puppetlabs/trapperkeeper/services/protocols/filesystem_watch_service.clj

```diff
@@ -4,8 +4,11 @@
 
 (def Event
   "Schema for an event on a file watched by this service."
-  {:type (schema/enum :create :modify :delete :unknown)
-   :path File})
+  (schema/if #(= (:type %) :unknown)
+    {:type (schema/eq :unknown)
+     :path (schema/pred nil?)}
```

Comment:
`(schema/pred nil?)` seems very strange to me, I guess that matches any value or `nil`, huh?  I've used `Object` for that in the past but I'm not sure if either one is better.  The best solution seems like it would probably be to just omit the `:path` key for `:unknown` events.